scan/lib/scan/runner.rb

```diff
@@ -94,17 +94,24 @@ def handle_results(tests_exit_status)
       end
     end
 
-    def zip_build_products
+    def self.zip_build_products_path
       return unless Scan.config[:should_zip_build_products]
 
-      # Gets :derived_data_path/Build/Products directory for zipping zip
-      derived_data_path = Scan.config[:derived_data_path]
-      path = File.join(derived_data_path, "Build/Products")
-
       # Gets absolute path of output directory
       output_directory = File.absolute_path(Scan.config[:output_directory])
       output_path = File.join(output_directory, "build_products.zip")
 
+      return output_path
+    end
+
+    def zip_build_products
+      output_path = Scan::Runner.zip_build_products_path
+      return unless output_path
```

Comment:
Isn't that a bit more cumbersome and harder to understand than just to have `return unless Scan.config[:should_zip_build_products]` here once again, and only then get the output_path?

Reply:
Ehhhh, yes and no. I didn't want `Scan.config[:should_zip_build_products]` to have to instances where it was getting looked up but I can see why this might not be ideal either 🤔

Reply:
I will rework